lib/spack/spack/util/spack_yaml.py

```diff
@@ -60,10 +65,56 @@ class syaml_str(str):
     __repr__ = str.__repr__
 
 
+class syaml_int(int):
+    __repr__ = str.__repr__
+
+
+#: primitive type -> syaml type mapping
```

Comment:
I think this should be reversed: `syaml_type -> primitive type mapping`

Reply:
This is a map *from* syaml type *to* primitive type. The comment is confusing as is (or did you disagree with that?)